describe/describe.py

```diff
@@ -2,133 +2,244 @@
+    :param update_dataset:          when the table is a registered dataset update the charts in-place
+    :param frac:                    when the table has more than 5000 samples,
+                                    the function will execute on random frac from the data (0.1 default)
+    """
+
     df = table.as_df()
-    header = df.columns.values
+    if df.shape[0] > MAX_SIZE_OF_DF:
+        df = df.sample(frac=frac)
     extra_data = {}
 
+    _create_histogram_mat_artifact(
+        context, df, extra_data, label_column, plots_dest
+    )
+    _create_features_histogram_artifacts(
+        context, df, extra_data, label_column, plots_dest
+    )
+    _create_violin_artifact(
+        context, df, extra_data, plots_dest
+                            )
+    _create_imbalance_artifact(
+        context, df, extra_data, label_column, plots_dest
+                               )
+    _create_corr_artifact(
+        context, df, extra_data, label_column, plots_dest
+                          )
+
+    artifact = table._artifact_url
+    if artifact is None:  # dataset not stored
+        artifact = context.log_dataset(key="dataset", db_key=name, stats=True, df=df)
+
+    if update_dataset:
+        from mlrun.artifacts import update_dataset_meta
+
+        update_dataset_meta(artifact, extra_data=extra_data)
+
+    # TODO : Plots according to client wishes - like preform histogram on selected features.
+    # TODO : 3-D plot on on selected features.
+    # TODO : Reintegration plot on on selected features.
+
+
+def _create_histogram_mat_artifact(
+    context: MLClientCtx,
+    df: pd.DataFrame,
+    extra_data: dict,
+    label_column: str,
+    plots_dest: str,
+):
+    """
+    Create and log a histogram matrix artifact
+    """
     try:
-        gcf_clear(plt)
-        snsplt = sns.pairplot(df, hue=label_column)  # , diag_kws={"bw": 1.5})
-        extra_data["histograms"] = context.log_artifact(
-            PlotArtifact("histograms", body=plt.gcf()),
-            local_path=f"{plots_dest}/hist.html",
-            db_key=False,
+        df_new = df.copy()
+        df_new[label_column] = df_new[label_column].apply(str)
+        fig = ff.create_scatterplotmatrix(
+            df_new, diag="histogram", index=label_column, width=2500, height=2500
+        )
+        fig.update_layout(title_text="<i><b>Histograms matrix</b></i>")
+        extra_data["histogram matrix"] = context.log_artifact(
+            PlotlyArtifact(key="histograms matrix", figure=fig),
+            local_path=f"{plots_dest}/hist_mat.html",
         )
+    except Exception as e:
+        context.logger.error(f"Failed to create histogram matrix artifact due to: {e}")
+
+
+def _create_features_histogram_artifacts(
+    context: MLClientCtx,
+    df: pd.DataFrame,
+    extra_data: dict,
+    label_column: str,
+    plots_dest: str,
+):
+    """
+    Create and log a histogram artifact for each feature
+    """
+    try:
```

Comment:
The function should not monitor itself with try-except. The caller should :)
Its ok to use try-except in a function, but when its the entire function, it should happen outside of it.